File: Python/Algorithm/programmers/caesar.py
# [caesar.py]
# 시저암호

def solution(s, n):
    alphabet = ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"]
    answer = ''

    for i in s:
        if i == " ":
            answer += i
        elif i.isupper():
            answer += alphabet[(alphabet.index(i.lower())+n) % len(alphabet)].upper()
        else:
            answer += alphabet[(alphabet.index(i.lower())+n) % len(alphabet)]
    return answer

File: Java/Algorithm/programmers/caesar.java
// [caesar.java]
// 시저 암호

class Solution {
  public String solution(String s, int n) {
      String answer = "";

      // 1. 한글자씩 반복
      for (int i = 0; i < s.length(); i++) {
          char ch = s.charAt(i);
          
          // 1-1. 공백일 경우
          if (Character.isWhitespace(ch)) {
              answer += " ";
              continue;
          // 1-2. 대문자와 소문자에 따라 아스키코드 계산 후 문자변환
          } else if (Character.isUpperCase(ch)){
              answer += (char)((ch - 'A' + n) % 26 + 'A');
          } else {
              answer += (char)((ch - 'a' + n) % 26 + 'a');
          }
      }
      return answer;
  }
}